src/index.tsx

```diff
@@ -90,26 +88,22 @@ export class RCPhoneInput extends React.Component<IProps, IState> {
   }
 
   public componentDidMount() {
-    const { onlyCountries, withIpLookup, onChange } = this.props
-
-    if (withIpLookup) {
-      fetchJsonp('https://ipinfo.io')
-      .then(response => response.json())
-      .then(json => {
-        const selectedCountry = onlyCountries.find(country => country.iso2 === json.country.toLowerCase())
-        const highlightCountryIndex = allCountries.findIndex(item => item === selectedCountry)
-
-        if (selectedCountry && highlightCountryIndex) {
-          if (typeof onChange === 'function') {
-            onChange({country: selectedCountry, number: ''})
-          }
+    const { onlyCountries, countryCode, onChange } = this.props
 
-          this.setState({
-            selectedCountry,
-            highlightCountryIndex
-          })
+    if (countryCode) {
+      const selectedCountry = onlyCountries.find(country => country.iso2 === countryCode.toLowerCase())
+      const highlightCountryIndex = allCountries.findIndex(item => item === selectedCountry)
+
+      if (selectedCountry && highlightCountryIndex) {
```

Comment:
Can we do?
```js
if (! (selectedCountry && highlightCountryIndex) || typeof onChange !== 'function') {
  return
}
```

Reply:
@antonversal not really, a `return` will prevent `setState` call when  `onChange` is not defined